doc/arch/adr-007-pa-dep-client-app.md

```diff
@@ -0,0 +1,122 @@
+this workflow does not fit neatly in MMW,
+which always expects a single area of interest.
+Rather than change MMW to accomodate this unusual and rare use case,
+it would be much simpler to create a new client app that is focused on this workflow.
+It will use the MMW API for all calculations.
+
+Given that we'll have a new client app,
+this is an opportunity to introduce a more modern tech stack on the front-end.
+This was recently done for [a similar project][beekeepers],
+where a new front-end stack was added within a Django app to sandbox it from the rest of the project.
+This document aims to discuss the pros and cons of different arrangements for the new client app,
+and pick one that satisfies all project constraints.
+
+The one major directive from the clients is that,
+from a user's perspective,
+the site should be deeply integrated with MMW.
+That is to say,
+it should use the same colors and branding,
+and not feel like its own project.
+
+From the development perspective,
+we should aim for using the latest front-end stacks,
+clean communication with the back-end API,
+and having a fast deployment pipeline.
+
+### Repository: Same vs Different
+
+Advantages of having the **same repository**:
+
+  * Familiarity, as it was done in the [Beekeepers][beekeepers] project
+  * All issues can be in one repo, simplifying project management
+  * Tasks that affect both front-end and back-end can be tracked in one commit or PR
+  * Simpler deployment, as front-end and back-end will always be in sync
+  * Styles can be easily reused between the two apps
+  * Easier to simulate production environments in development,
+    as we only need to start a single development environment.
+
+Advantages of having **different repositories**:
+
+  * Cleaner logical separation in the codebase
+  * Fewer cohabiting developer entry points, i.e. `package.json`, or multiple `node_modules`
+  * Does not require changing the `node` or `npm` version in MMW
+  * Does not require adding a Docker environment in MMW for building the front-end
+  * Faster deployment, as the client app can be deployed independently of the main app
+
+### Deployment: Paired vs Independent
+
+Advantages of **paired deployment**:
+
+  * Front-end and back-end will always be in sync
+  * Does not require extra configuration of AWS
+  * Does not need extra steps during deployment
+  * Easier to write end-to-end tests
+
+Advantages of **independent deployment**:
+
+  * Front-end can be iterated much faster, without the heavy deployment cycle of MMW
+
+### Access: Subdomain vs Subdirectory
+
+Advantages of **subdomain access**:
+
+  * More easily supports independent deployments
+
+Advantages of **subdirectory access**:
+
+  * Maintains MMW brand identity
+  * Would not have any CORS issues
+    - This would not apply to the documented public API endpoints,
+      but there are some undocumented APIs such as MapShed which we'll need to run,
+      which may have CORS headers set on them.
```

Comment:
We could add MapShed endpoints to the public API as part of this work.

Reply:
That is not an explicit requirement of this contract, and adding the documentation needed for it may be a considerable effort. Furthermore, exposing those endpoints may have traffic implications. Furthermore, there may be business reasons to keep them undocumented at this time (potential monetization, for one).

These APIs are undocumented, but not inaccessible, so we the experts should be able to work with them without problems.